packages/react-native-performance/android/src/main/java/com/oblador/performance/RNPerformance.java

```diff
@@ -0,0 +1,141 @@
+    }
+
+    private static class LoadRNPerformance {
+        static final RNPerformance instance = new RNPerformance();
+    }
+
+    public static RNPerformance getInstance() {
+        return LoadRNPerformance.instance;
+    }
+
+    interface MarkerListener {
+        void logMarker(PerformanceEntry mark);
+    }
+
+    private static final List<MarkerListener> sListeners = new CopyOnWriteArrayList<>();
+    private final List<PerformanceEntry> entries = new CopyOnWriteArrayList<>();
+
+    @DoNotStrip
+    protected void addListener(MarkerListener listener) {
+        if (!sListeners.contains(listener)) {
+            sListeners.add(listener);
+        }
+    }
+
+    @DoNotStrip
+    protected void removeListener(MarkerListener listener) {
+        if (!sListeners.contains(listener)) {
+            sListeners.remove(listener);
+        }
+    }
+
+    public void mark(String markName) {
+        mark(markName, true);
+    }
+
+    public void mark(String markName, boolean ephemeral) {
+        mark(markName, null, ephemeral);
+    }
+
+    public void mark(String markName, Bundle detail) {
+        mark(markName, detail, true);
+    }
+
+    public void mark(String markName, Bundle detail, boolean ephemeral) {
+        PerformanceEntry mark = new PerformanceMark(
+                markName,
+                System.currentTimeMillis(),
+                ephemeral,
+                detail
+        );
+        addEntry(mark);
+    }
+
+    public void metric(String metricName, double value) {
+        metric(metricName, value, true);
+    }
+
+    public void metric(String metricName, double value, boolean ephemeral) {
+        metric(metricName, value, null, ephemeral);
+    }
+
+    public void metric(String metricName, double value, Bundle detail) {
+        metric(metricName, value, detail, true);
+    }
+
+    public void metric(
+            String metricName,
+            double value,
+            Bundle detail,
+            boolean ephemeral
+    ) {
+        PerformanceEntry mark = new PerformanceMetric(
+                metricName,
+                value,
+                System.currentTimeMillis(),
+                ephemeral,
+                detail
+        );
+        addEntry(mark);
+    }
+
+    private void addEntry(PerformanceEntry mark) {
+        entries.add(mark);
+        emitMark(mark);
+    }
+
+    private void emitMark(PerformanceEntry mark) {
+        for (MarkerListener listener : sListeners) {
+            listener.logMarker(mark);
+        }
+    }
+
+    protected List<PerformanceEntry> getEntries() {
+        return entries;
+    }
+
+    protected void clearEntries() {
+        entries.clear();
+    }
+
+    protected void clearEntries(String name) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            entries.removeIf((entry) -> entry.getName().equals(name));
+        } else {
+            for (PerformanceEntry entry : entries) {
+                if (entry.isEphemeral()) {
```

Comment:
Looks like a copy/paste accident: I guess this should be `if (entry.getName().equals(name))`.
Side note: I would personally stick with just a single implementation (the one compatible with all Android versions).

Reply:
Nice catch

Reply:
I think we can use the newer API of removeIf which can be trusted to be optimized after Android N.